src/RecipeMapper.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+class cRecipeMapper
+{
+public:
+	cRecipeMapper(void);
+	~cRecipeMapper();
+	UInt32 recipeNameToRecipeID(const AString & aRecipeName);
+	AString recipeIDToRecipeName(UInt32 recipeId);
+
+private:
+	std::map<UInt32, AString> m_Protocol1_12_2;
```

Comment:
Everything outside of the protocol folder should really be protocol version independent. We should use a consistent ID everywhere internally and only convert it at the last second in the protocol handlers.

Reply:
i.e. have a "string -> internal id" mapping and an "internal id -> protocol id". Use internal ids wherever possible.

Reply:
Ok, so I will move `RecipeMapper` into the `Protocol` directory.

Reply:
I don't particularly care about what folder `cRecipeMapper` is in. I care about what code the protocol-specific IDs are used in. You are using these IDs in code that is meant to be protocol-agnostic.

Reply:
K, get it, thx